feature-libs/product-multi-dimensional/selector/components/selector/product-multi-dimensional-selector.component.ts

```diff
@@ -42,14 +42,17 @@ export class ProductMultiDimensionalSelectorComponent {
     CurrentProductService
   );
 
-  product$: Observable<Product> = this.currentProductService.getProduct().pipe(
-    filter(isNotNullable),
-    distinctUntilChanged(),
-    shareReplay(1),
-    tap((product) => {
-      this.categories = this.getVariants(product);
-    })
-  );
+  product$: Observable<Product> = this.currentProductService
+    .getProduct(ProductScope.MULTI_DIMENSIONAL)
+    .pipe(
+      filter(isNotNullable),
+      filter((product: Product) => !!product?.multidimensional),
+      distinctUntilChanged(),
+      shareReplay(1),
+      tap((product: Product) => {
+        this.categories = this.getVariants(product);
+      })
+    );
```

Comment:
```suggestion
  product$: Observable<Product> = this.currentProductService
    .getProduct(ProductScope.MULTI_DIMENSIONAL)
    .pipe(
      filter(isNotNullable),
      filter((product) => !!product.multidimensional),
      distinctUntilChanged(),
      shareReplay(1),
      tap((product) => {
        this.categories = this.getVariants(product);
      })
    );
```
Types are not required due to type narrowing. In addition, you don't need to use optional chaining as at this point we know `product` is not null.
Apart from that, I'm leaving an alternative usage of the `filter` operator that combines two conditions:
```
filter((product): product is Product => isNotNullable(product) && !!product.multidimensional),
```
Checking the condition during one iteration might be more performant, but I'm not sure whether we suffer on the performance in this place.

Reply:
TypeScript doesn't like this `filter((product) => !!product.multidimensional),` which is why I've added the check. Also, it breaks the library build if left without the null check. In general, JavaScript is rather fast, so I don't think it makes any difference.